tests/collections/asr/decoding/test_rnnt_decoding.py

```diff
@@ -0,0 +1,252 @@
+
+        input_signal = torch.tensor(audio, dtype=torch.float32).unsqueeze(0)
+        input_signal_length = torch.tensor([len(audio)], dtype=torch.int32)
+
+        encoded, encoded_len = model(input_signal=input_signal, input_signal_length=input_signal_length)
+
+    return model, encoded, encoded_len
+
+
+def decode_text_from_greedy_hypotheses(hyps, decoding):
+    decoded_hyps = decoding.decode_hypothesis(hyps)  # type: List[str]
+
+    return decoded_hyps
+
+
+def decode_text_from_nbest_hypotheses(hyps, decoding):
+    hypotheses = []
+    all_hypotheses = []
+
+    for nbest_hyp in hyps:  # type: rnnt_utils.NBestHypotheses
+        n_hyps = nbest_hyp.n_best_hypotheses  # Extract all hypotheses for this sample
+        decoded_hyps = decoding.decode_hypothesis(n_hyps)  # type: List[str]
+
+        hypotheses.append(decoded_hyps[0])  # best hypothesis
+        all_hypotheses.append(decoded_hyps)
+
+    return hypotheses, all_hypotheses
+
+
+class TestRNNTDecoding:
+    @pytest.mark.unit
+    def test_constructor(self):
+        cfg = RNNTDecodingConfig()
+        vocab = char_vocabulary()
+        decoder = get_rnnt_decoder(vocab_size=len(vocab))
+        joint = get_rnnt_joint(vocab_size=len(vocab))
+        decoding = RNNTDecoding(decoding_cfg=cfg, decoder=decoder, joint=joint, vocabulary=vocab)
+        assert decoding is not None
+
+    @pytest.mark.unit
+    def test_constructor_subword(self, tmp_tokenizer):
+        cfg = RNNTBPEDecodingConfig()
+        vocab = tmp_tokenizer.vocab
+        decoder = get_rnnt_decoder(vocab_size=len(vocab))
+        joint = get_rnnt_joint(vocab_size=len(vocab))
+        decoding = RNNTBPEDecoding(decoding_cfg=cfg, decoder=decoder, joint=joint, tokenizer=tmp_tokenizer)
+        assert decoding is not None
+
+    @pytest.mark.skipif(
+        not NUMBA_RNNT_LOSS_AVAILABLE, reason='RNNTLoss has not been compiled with appropriate numba version.',
+    )
+    @pytest.mark.with_downloads
+    @pytest.mark.unit
+    def test_greedy_decoding_preserve_alignments(self, test_data_dir):
+        model, encoded, encoded_len = get_model_encoder_output(test_data_dir, 'stt_en_conformer_transducer_small')
+
+        beam = greedy_decode.GreedyRNNTInfer(
+            model.decoder,
+            model.joint,
+            blank_index=model.joint.num_classes_with_blank - 1,
+            max_symbols_per_step=5,
+            preserve_alignments=True,
+        )
+
+        enc_out = encoded
+        enc_len = encoded_len
+
+        with torch.no_grad():
+            hyps = beam(encoder_output=enc_out, encoded_lengths=enc_len)[0]  # type: rnnt_utils.Hypothesis
+            hyp = decode_text_from_greedy_hypotheses(hyps, model.decoding)
+            hyp = hyp[0]
+
+            assert hyp.alignments is not None
+
+            # Use the following commented print statements to check
+            # the alignment of other algorithms compared to the default
+            # print("Text", hyp.text)
```

Comment:
I usually prefer to leave the print statements "on" all the time - by default test output is not being printed, but if something fails, it's very helpful to see such messages right away without the need to change code and rerun

Reply:
Agreed, made them print always